fiatconnect-api.md

```diff
@@ -383,105 +406,241 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
 the second authenticates the *client*, and is optionally required, depending on whether or not it has been configured by the client
 with the provider.
 
-### 3.2.1. JWT Authentication
+### 3.2.1. Sign-In With Ethereum
 
-Only users with a registered address on the Celo blockchain may make requests to a FiatConnect API. The FiatConnect API specification
-requires servers to recognize a JWT token signed with the user's private key, and verifiable with the user's public key. With this authentication
-scheme, a server can:
+FiatConnect-compliant APIs are required to use the *Sign-In With Ethereum* standard, otherwise known as [EIP-4361](https://eips.ethereum.org/EIPS/eip-4361)
+in order to identify users and confirm ownership of a Celo blockchain address (private/public keypair). At a high level, the Sign-In With Ethereum standard
+requires the client to use a user's private key to sign a plaintext message containing a number of standardized fields, and send that signed message to the
+server to prove account ownership. FiatConnect APIs MUST use the SIWE standard to create authenticated sessions for the user, and return session cookies to
+the client. More detail on exactly how this should be structured is provided below.
 
-1. Associate a user's Celo blockchain address with the request, and
-2. Verify the requester's ownership of the provided address.
+#### 3.2.1.1. SIWE Sessions
 
-Servers MUST parse and recognize JWT tokens according to the following format. Servers MUST validate these tokens on every request.
+At a high level, the client will initiate a login request with a server by requesting a SIWE-compliant message to sign from a `GET /auth/login` endpoint. This
```

Comment:
Rather than having the client create the message from scratch, I'm proposing a scheme where the server issues a "challenge" to the client in the form of a message to sign. This is more idiomatic, and is somewhat "baked into" the SIWE spec; the `${domain} wants you to sign in with your Ethereum account: ${address}` template is required in _all_ SIWE messages, implying that the message is meant to be issued by the `domain` itself. This also means the client and server can coordinate their clocks, which is actually pretty handy, especially considering the additional security constraints imposed on the transfer endpoints (described below) where clock coordination is relevant.

Reply:
sounds like this is outdated now

Reply:
so I'm pressing "resolve"